mirantis/testing/metta_states/state_environment.py

```diff
@@ -105,15 +105,16 @@ def set_state(self, state: str):
             state_fixture: Fixture = self._fixtures.get(
                 instance_id=state, interfaces=[METTA_PLUGIN_INTERFACE_ROLE_ENVIRONMENTSTATE]
             )
-            self._active_state_id = state_fixture.instance_id
-
-            # activate the state plugin
-            if hasattr(state_fixture.plugin, "activate"):
-                state_fixture.plugin.activate()
 
         except KeyError as err:
             raise RuntimeError(f"Unknown State requested for activation: {state}") from err
 
+        self._active_state_id = state_fixture.instance_id
+
+        # activate the state plugin
+        if hasattr(state_fixture.plugin, "activate"):
+            state_fixture.plugin.activate()
+
```

Comment:
this block was moved out of the exception catch to prevent mis-diagnosis of exceptions raised during activation.